app/src/main/java/com/example/sharewhatyoucanproject/LocationTrack.kt

```diff
@@ -0,0 +1,250 @@
+    // to handle the case where the user grants the permission. See the documentation
+    // for ActivityCompat#requestPermissions for more details.
+
+
+    /*if (checkNetwork) {
+
+
+           if (ActivityCompat.checkSelfPermission(mContext, Manifest.permission.ACCESS_FINE_LOCATION) != PackageManager.PERMISSION_GRANTED && ActivityCompat.checkSelfPermission(mContext, Manifest.permission.ACCESS_COARSE_LOCATION) != PackageManager.PERMISSION_GRANTED) {
+               // TODO: Consider calling
+               //    ActivityCompat#requestPermissions
+               // here to request the missing permissions, and then overriding
+               //   public void onRequestPermissionsResult(int requestCode, String[] permissions,
+               //                                          int[] grantResults)
+               // to handle the case where the user grants the permission. See the documentation
+               // for ActivityCompat#requestPermissions for more details.
+           }
+           locationManager.requestLocationUpdates(
+                   LocationManager.NETWORK_PROVIDER,
+                   MIN_TIME_BW_UPDATES,
+                   MIN_DISTANCE_CHANGE_FOR_UPDATES, this);
+
+           if (locationManager != null) {
+               loc = locationManager
+                       .getLastKnownLocation(LocationManager.NETWORK_PROVIDER);
+
+           }
+
+           if (loc != null) {
+               latitude = loc.getLatitude();
+               longitude = loc.getLongitude();
+           }
+       }*/
+// if GPS Enabled get lat/long using GPS Services
+    // get GPS status
+
+    // get network provider status
+    private val location: Location?
+        private get() {
+            try {
+                locationManager = mContext
+                    .getSystemService(LOCATION_SERVICE) as LocationManager
+
+                // get GPS status
+                checkGPS = locationManager!!
+                    .isProviderEnabled(LocationManager.GPS_PROVIDER)
+
+                // get network provider status
+                checkNetwork = locationManager!!
+                    .isProviderEnabled(LocationManager.NETWORK_PROVIDER)
+                if (!checkGPS && !checkNetwork) {
+                    Toast.makeText(mContext, "No Service Provider is available", Toast.LENGTH_SHORT)
+                        .show()
+                } else {
+                    canGetLocation = true
+
+                    // if GPS Enabled get lat/long using GPS Services
+                    if (checkGPS) {
+                        if (ActivityCompat.checkSelfPermission(
+                                mContext,
+                                permission.ACCESS_FINE_LOCATION
+                            ) != PackageManager.PERMISSION_GRANTED && ActivityCompat.checkSelfPermission(
+                                mContext, permission.ACCESS_COARSE_LOCATION
+                            ) != PackageManager.PERMISSION_GRANTED
```

Comment:
This is hard to read. I have no idea what's going on here.  Could we break this up, so it's easier to follow?